### PR TITLE
Clean up

### PR DESCRIPTION
**Pull request by valfirst**
## Change list

- Remove redundant 'throws' clauses
- Simplify assertions in unit tests
- Replace 'Collection.addAll()' with parametrized constructor call
- Remove redundant array creation for calling varargs method
- Remove redundant type arguments
 
## Types of changes

- [X] No changes in production code.
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: src/main/java/io/appium/java_client/pagefactory/bys/builder/AppiumByBuilder.java [hunk 1]
```diff
@@ -96,7 +96,7 @@ private static String getFilledValue(Annotation mobileBy) {
             }
 
             try {
-                String strategyParameter = value.invoke(mobileBy, new Object[] {}).toString();
+                String strategyParameter = value.invoke(mobileBy).toString();
```
Comment (mykola-mokhnach) [inline src/main/java/io/appium/java_client/pagefactory/bys/builder/AppiumByBuilder.java]: does this call return the same result as the previous one?

File: src/main/java/io/appium/java_client/pagefactory/bys/builder/AppiumByBuilder.java [hunk 2]
```diff
@@ -96,7 +96,7 @@ private static String getFilledValue(Annotation mobileBy) {
...
                 if (!"".equals(strategyParameter)) {
```
Comment (mykola-mokhnach) [inline src/main/java/io/appium/java_client/pagefactory/bys/builder/AppiumByBuilder.java]: we could use isEmpty() here

File: src/test/java/io/appium/java_client/ios/IOSDriverTest.java [hunk 3]
```diff
@@ -138,6 +138,6 @@ public void getDeviceTimeTest() {
         driver.toggleTouchIDEnrollment(true);
         driver.performTouchID(true);
         driver.performTouchID(false);
-        assertEquals(true, true);
+        assertTrue(true);
```
Comment (mykola-mokhnach) [inline src/test/java/io/appium/java_client/ios/IOSDriverTest.java]: %)

File: src/test/java/io/appium/java_client/service/local/ThreadSafetyTest.java [hunk 4]
```diff
@@ -86,8 +87,8 @@
                 throw isRunningTestThread2.throwable;
             }
 
-            assertTrue(isRunningTestThread.result.equals(true));
-            assertTrue(isRunningTestThread2.result.equals(true));
+            assertEquals(true, isRunningTestThread.result);
```
Comment (mykola-mokhnach) [inline src/test/java/io/appium/java_client/service/local/ThreadSafetyTest.java]: cannot we just test that `assertTrue(isRunningTestThread.result)` ?

File: src/main/java/io/appium/java_client/pagefactory/bys/builder/AppiumByBuilder.java (hunk 1 above)
Comment (valfirst) [reply src/main/java/io/appium/java_client/pagefactory/bys/builder/AppiumByBuilder.java]: yes, the method signature is:
```java
java.lang.reflect.Method.invoke(Object, Object...)
```
so it's not required to pass an empty array as varargs

File: src/test/java/io/appium/java_client/service/local/ThreadSafetyTest.java (hunk 4 above)
Comment (valfirst) [reply src/test/java/io/appium/java_client/service/local/ThreadSafetyTest.java]: we can't: `assertTrue` requires `boolean` input parameter, but `isRunningTestThread.result` is of `Object` type

Comment (mykola-mokhnach) [reply src/test/java/io/appium/java_client/service/local/ThreadSafetyTest.java]: I'd rather use Boolean.valueOf then

File: src/test/java/io/appium/java_client/ios/IOSDriverTest.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (mykola-mokhnach) [reply src/test/java/io/appium/java_client/ios/IOSDriverTest.java]: yep, now I remember why `assertEquals(true, true);` has been used ;)

**Pull request merged**